src/metrics/matcher/match.go

```diff
@@ -28,10 +28,15 @@ import (
 	"github.com/m3db/m3/src/metrics/matcher/cache"
 	"github.com/m3db/m3/src/metrics/metric/id"
 	"github.com/m3db/m3/src/metrics/rules"
+	"github.com/m3db/m3/src/metrics/rules/view"
 )
 
 // Matcher matches rules against metric IDs.
 type Matcher interface {
+	// LatestRollupRules returns the latest rollup rules for a given namespace
+	// at a given time.
+	LatestRollupRules(namespace []byte, timeNanos int64) ([]view.RollupRule, error)
```

Comment:
what is going to be calling this? I have a pretty large change in flight that is going to make the `Matcher` not thread safe, so one will need to be created per request.

Reply:
I was hoping to be able to call this in middleware, on the query path.

Reply:
k. we probably don't want to expose this through the Downsampler then, since it's a global object. Instead each query request would create a new matcher (or get one from a pool)

Reply:
OK let me look into this.

Reply:
for now just expose through the Downsampler to keep things consistent. When we land the refactor to make Matchers request scoped, we can clean this code up too. not a big deal.

Reply:
Will do!